### PR TITLE
Fixed ##1369: Set up front end testing environment with Nextjs

### LINKED ISSUE #1369: [next] Port front-end tests to work with Next.js
**What would you like to be added**:
[Port our front-end tests to use Next.js style](https://github.com/Seneca-CDOT/telescope/issues/1316)
**Why would you like this to be added**:
Next.js Migration

### PR DESCRIPTION
**Pull request by tonyvugithub**
## Issue This PR Addresses
Fixed #1369 

## Type of Change

- [ ] **Bugfix**: Change which fixes an issue
- [x] **New Feature**: Change which adds functionality
- [ ] **Documentation Update**: Change which improves documentation
- [ ] **UI**: Change which improves UI

## Description
Set up frontend testing environment with Nextjs

Screenshot of successful sample test
image

## Checklist

- [x] **Quality**: This PR builds and passes our npm test and works locally
- [x] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [x] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not (if applicable)
- [ ] **Documentation**: This PR includes updated/added documentation to user exposed functionality or configuration variables are added/changed or an explanation of why it does not(if applicable)

### CODE REVIEW

File: src/frontend/package.json [hunk 1]
```diff
@@ -6,20 +6,29 @@
     "develop": "next dev -p 8000",
     "build": "next build",
     "start": "next start",
+    "test": "jest --watchAll",
```
Comment (humphd) [inline src/frontend/package.json:9]: We don't want to use watch when running `test`, it needs to be a separate command so that CI doesn't hang forever.

**Review by humphd — CHANGES_REQUESTED** [review]

**Comment by tonyvugithub** [comment]
@humphd : I can't make it work. After spending the whole day trying countless solutions that I can find from the internet. It just does not run probably. I talked to @manekenpix about this and we even tried to make changes to circleci config file to get it passed. turned out there might be some conflicts in the settings that confuse the app. My frontend setup runs just fine if I manually CD to the frontend directory and run `npm run test ` the test suite run. But if I run npm run `test-frontend` from root, the test suites did not run and gave me the below error. We suspect that having the frontend in a different repo might solve the issue we are having. 
image

**Comment by humphd** [comment]
Personally,  I think we should hold off on this.  I think we need to split the next.js fronte-end stuff into a separate repo and then we can do things without cross-polluting the configs for the backend.

**Comment by tonyvugithub** [comment]
@humphd : That was what we concluded yesterday too. OK then I just close this then? If it will be on another repo

**Comment by humphd** [comment]
Yeah, let's close for now and follow-up when we move to a new repo layout.

**Pull request closed without merging**